### DISCUSSION TITLE
Spell changes from build 13930

### DISCUSSION THREAD

**Discussion by Ghaster** [TrinityCore/TrinityCore#27613] [General]
As some may know, China was a bit late to the whole WotLK thing. This meant they actually ended the expansion with a significantly newer build compared to the rest of the world. This build is known as 3.3.5b.13930 and is dated Apr 22 2011, making it the very last and newest build of WotLK.

While the changes between 12340 and 13930 are quite minimal, there are some differences in the shipped database files. I have listed all changes between the two's shipped spell database as an ApplySpellFix, separated into changes that also exist in later versions of the game and those unique to 13930. This should allow the changes to be easily applied to the 3.3.5 branch.

I am creating this as a discussion, as these changes should not be applied blindly. Some of the issues that these changes fix might not have been actual behavior for 12340 and were only fixed after the rest of the world switched to 4.0.1.

# Changes that exists in newer versions

### Ez-Thro Dynamite
```
// Ez-Thro Dynamite
ApplySpellFix({ 8331, 23000 }, [](SpellInfo* spellInfo)
{
    spellInfo->_GetEffect(EFFECT_1).Effect = SPELL_EFFECT_NONE;
    spellInfo->_GetEffect(EFFECT_1).TargetA = SpellImplicitTargetInfo();
    spellInfo->_GetEffect(EFFECT_1).RadiusEntry = nullptr;
    spellInfo->_GetEffect(EFFECT_1).TriggerSpell = 0;
});
```
This one is quite interesting. Apparently, at some point during 3.1, a feature was added to throw Ez-Thro Dynamite into fishing pools ([https://web.archive.org/web/20130706020115/http://www.elsprofessions.com:80/news/dynamite-fishing.html](url)). However, it was removed at some point, but the exact date is debatable. Some sources are claiming this was removed in 4.0.1, while others state this was removed during 3.3.5. Whatever the case, this change in 13930 definitely removes it completely, as allowing level 10-19 getting easy access to Outland-era fishing materials sounds like an exploit waiting to happen.

### Swimming Speed
```
// Swimming Speed
ApplySpellFix({ 8747 }, [](SpellInfo* spellInfo)
{
    spellInfo->DurationEntry = nullptr;
});
```
Confirmed in later versions. This is a spell used by the item Azure Silk Belt ([https://www.wowhead.com/item=7052/azure-silk-belt](url)). Having the effect only active for a couple of seconds would be pretty sad.

### Unfriendly With Test
```
// Unfriendly With Test
ApplySpellFix({ 42806 }, [](SpellInfo* spellInfo)
{
    spellInfo->DurationEntry = sSpellDurationStore.LookupEntry(21);
    spellInfo->_GetEffect(EFFECT_0).DieSides = 0;
});
```
Confirmed in later versions. This is just some kind of debug spell, not much to say.

### Deepdiving Pearl
```
// Deepdiving Pearl
ApplySpellFix({ 44654 }, [](SpellInfo* spellInfo)
{
    spellInfo->_GetEffect(EFFECT_2).DamageMultiplier = 1.0f;
});
```
Confirmed in later versions. This simply sets an incorrect default value for that field to the correct one.

### Elekk Dispersion Ray - Kill Credit
```
// Elekk Dispersion Ray - Kill Credit
ApplySpellFix({ 44654 }, [](SpellInfo* spellInfo)
{
    spellInfo->_GetEffect(EFFECT_0).TargetA = SpellImplicitTargetInfo(TARGET_UNIT_CHANNEL_TARGET);
});
```
Confirmed in later versions. Spell is used for the pink elekk Brewfest quest. This fixes a bug allowing you to target one elekk and switch to another during channeling (described in WoWHead comments here: [https://www.wowhead.com/item=32960/elekk-dispersion-ray#comments](url)).

### Glacial Strike
```
// Glacial Strike
ApplySpellFix({ 71316, 71317 }, [](SpellInfo* spellInfo)
{
    spellInfo->AttributesEx3 |= SPELL_ATTR3_ONLY_TARGET_PLAYERS;
});
```
Confirmed in later versions. A spell used by a trash mob in ICC. Comment on WoWHead describes mind-controlling the trash mob and casting this spell on other mobs ([https://www.wowhead.com/npc=37228/frostwarden-warrior#comments](url)), this change might be a hotfix for that behavior.

### Playback Speech
```
// Playback Speech
ApplySpellFix({ 74209 }, [](SpellInfo* spellInfo)
{
    spellInfo->AttributesEx3 |= SPELL_ATTR3_UNK28;
});
```
Confirmed in later versions. In the leaked attributes, this unknown attribute means "Ignore Caster & Target Restrictions". Is related to Operation Gnomeregan event so most likely a bug fix.

### Fiery Combustion/Soul Consumption
```
ApplySpellFix({
    74562, // Fiery Combustion
    74792  // Soul Consumption
}, [](SpellInfo* spellInfo)
{
    spellInfo->AttributesEx |= SPELL_ATTR1_CANT_BE_REDIRECTED | SPELL_ATTR1_CANT_BE_REFLECTED;
});
```
Confirmed in later versions. Most likely added to prevent players from redirecting and reflecting Halion's abilities.

### Consumption
```
// Consumption
ApplySpellFix({ 75875 }, [](SpellInfo* spellInfo)
{
    spellInfo->_GetEffect(EFFECT_0).Mechanic = MECHANIC_NONE;
    spellInfo->_GetEffect(EFFECT_1).Mechanic = MECHANIC_SNARE;
});
```
Confirmed in later versions. This fix is actually already implemented in SpellMgr, but keeping it here for completeness sake.

### [DND]Play
```
// [DND]Play
ApplySpellFix({ 77984 }, [](SpellInfo* spellInfo)
{
    spellInfo->Attributes |= SPELL_ATTR0_HIDE_IN_COMBAT_LOG;
});
```
Confirmed in later versions. For a "Do Not Display" spell it would be pretty contradictory showing it in the combat log.

# Unique 13930 Changes

## Preparation
```
ApplySpellFix({
    546,   // Water Walking
    1066,  // Aquatic Form
    1706,  // Levitate
    2983,  // Sprint
    3714,  // Path of Frost
    5697,  // Unending Breath
    7355,  // Stuck
    7840,  // Swim Speed
    8747,  // Swimming Speed
    11319, // Water Walking
    24347, // Master Angler
    24930, // Hallow's End Candy
    41273, // Deepdiving Pearl
    50809, // Murliver Oil
    64731, // Sea Turtle
}, [](SpellInfo* spellInfo)
{
    spellInfo->ExcludeCasterAuraSpell = 44521;
});

// Preparation
ApplySpellFix({ 44521 }, [](SpellInfo* spellInfo)
{
    spellInfo->_GetEffect(EFFECT_1).MiscValue = 1791;
    spellInfo->_GetEffect(EFFECT_1).MiscValueB = 1;
});
```
I have no idea about this change. My only guess is that players were finding ways out of battleground spawn points and riding around the map using swim speed effects. Does not exist in any other version.

### Unfriendly With Test
```
// Unfriendly With Test
ApplySpellFix({ 42806 }, [](SpellInfo* spellInfo)
{
    spellInfo->_GetEffect(EFFECT_0).BasePoints = 4;
    spellInfo->_GetEffect(EFFECT_0).ApplyAuraName = SPELL_AURA_MOD_FACTION;
    spellInfo->_GetEffect(EFFECT_0).MiscValue = 35;
    spellInfo->_GetEffect(EFFECT_0).MiscValueB = 1;
});
```
Yes it is THAT spell again. In 13930, this spell also includes a few changes not found in a newer version. 

### Twilight Pulse Periodic
```
// Twilight Pulse Periodic
ApplySpellFix({ 78861 }, [](SpellInfo* spellInfo)
{
    spellInfo->_GetEffect(EFFECT_0).ApplyAuraName = SPELL_AURA_DUMMY;
    spellInfo->_GetEffect(EFFECT_0).Amplitude = 0;
    spellInfo->_GetEffect(EFFECT_0).TriggerSpell = 0;
});
```
Guess they switched to a scripted implementation for Halion's Twilight Pulse at some point. This change does not exist in later versions and would also require a script in the core if implemented.

**Comment by funjoker** [comment]
Do you want to bring 13930 changes to 12340 ?

**Reply by Ghaster** [reply]
Not necessarily, but some of these changes might have been applied serverside during 12340.

**Reply by funjoker** [reply]
i see. Another question :D does 12340 natively support 13930 ?
If no, maybe someone likes to add a patch file to do so? ^^
